pkg/lib/operation/dbt/generate/profile/operation.go

```diff
@@ -0,0 +1,93 @@
+package profile
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/keboola/go-client/pkg/client"
+	"go.opentelemetry.io/otel/trace"
+	"gopkg.in/yaml.v3"
+
+	"github.com/keboola/keboola-as-code/internal/pkg/filesystem"
+	"github.com/keboola/keboola-as-code/internal/pkg/log"
+	"github.com/keboola/keboola-as-code/internal/pkg/telemetry"
+)
+
+type dependencies interface {
+	Fs() filesystem.Fs
+	Logger() log.Logger
+	StorageApiClient() client.Sender
+	Tracer() trace.Tracer
+}
+
+type Options struct {
+	TargetName string
+}
+
+const profilePath = "profiles.yml"
+
+func Run(ctx context.Context, opts Options, d dependencies) (err error) {
+	ctx, span := d.Tracer().Start(ctx, "kac.lib.operation.dbt.generate.profile")
+	defer telemetry.EndSpan(span, &err)
+
+	if !d.Fs().Exists(`dbt_project.yml`) {
+		return fmt.Errorf(`missing file "dbt_project.yml" in the current directory`)
+	}
+	file, err := d.Fs().ReadFile(filesystem.NewFileDef(`dbt_project.yml`))
+	if err != nil {
+		return err
+	}
+	configFile := make(map[string]interface{})
+	err = yaml.Unmarshal([]byte(file.Content), &configFile)
+	if err != nil {
+		return err
+	}
+	profileName, ok := configFile["profile"]
+	if !ok {
+		return fmt.Errorf(`configuration file "dbt_project.yml" is missing "profile"`)
+	}
+
+	logger := d.Logger()
+	targetUpper := strings.ToUpper(opts.TargetName)
+	profileDetails := map[string]interface{}{
+		"target": opts.TargetName,
+		"outputs": map[string]interface{}{
+			opts.TargetName: map[string]interface{}{
+				"account":   fmt.Sprintf("{{ env_var(\"DBT_%s_ACCOUNT\") }}", targetUpper),
+				"database":  fmt.Sprintf("{{ env_var(\"DBT_%s_DATABASE\") }}", targetUpper),
+				"password":  fmt.Sprintf("{{ env_var(\"DBT_%s_PASSWORD\") }}", targetUpper),
+				"schema":    fmt.Sprintf("{{ env_var(\"DBT_%s_SCHEMA\") }}", targetUpper),
+				"type":      fmt.Sprintf("{{ env_var(\"DBT_%s_TYPE\") }}", targetUpper),
+				"user":      fmt.Sprintf("{{ env_var(\"DBT_%s_USER\") }}", targetUpper),
+				"warehouse": fmt.Sprintf("{{ env_var(\"DBT_%s_WAREHOUSE\") }}", targetUpper),
+			},
```

Comment:
V originále se teda do názvu env variable strká název konfigurace (https://github.com/keboola/dbt-transformation/blob/b0422211138562e93f9319585161056722e1d9b2/src/Command/GenerateProfilesAndSourcesCommand.php#L218) ale tím že tady listování konfigurací neřešíme, tak bych tam dal název targetu, ok?

Reply:
jj, tiez som to tak pochopil, ked sme sa pytali na feature calls, adam tam ma `DBT_KBC_` my iba `DBT`, nechal by som tam aj to `KBC`, na odlisenie.

Reply:
On tam má právě napevno jen `DBT_` (https://github.com/keboola/dbt-transformation/blob/b0422211138562e93f9319585161056722e1d9b2/src/Command/GenerateProfilesAndSourcesCommand.php#L218) a to `KBC_` se tam potom dostane z názvu konfigurace (https://github.com/keboola/dbt-transformation/blob/d6dcbb71230569155ce4115d0dccb71eba8aad9d/src/Command/GenerateProfilesAndSourcesCommand.php#L95), ale upravím, není problém.

Reply:
aha ok, ja som nepozeral kod, ale iba testy ... tak su to nase ENVs, ktore nastavujeme my, takze by som to tam skor dal ako nedal (prevencia kolizii)